packages/mds-geography-author/api.ts

```diff
@@ -0,0 +1,293 @@
+import express from 'express'
+import db from '@mds-core/mds-db'
+
+import {
+  pathsFor,
+  ServerError,
+  NotFoundError,
+  DependencyMissingError,
+  ValidationError,
+  AlreadyPublishedError,
+  InsufficientPermissionsError,
+  BadParamsError
+} from '@mds-core/mds-utils'
+import { geographyValidationDetails } from '@mds-core/mds-schema-validators'
+import log from '@mds-core/mds-logger'
+
+import { checkAccess } from '@mds-core/mds-api-server'
+
+function api(app: express.Express): express.Express {
+  app.get(
+    pathsFor('/geographies/meta/'),
+    checkAccess(scopes => {
+      return scopes.includes('geographies:read:published') || scopes.includes('geographies:read:unpublished')
+    }),
+    async (req, res) => {
+      // TODO
+      // if get_unpublished is true, and client lacks geographies:read:published, then
+      // unpublished geos should still be filtered out
+      const { scopes } = res.locals
+      const { get_published = null, get_unpublished = null } = req.query
+      const params = { get_published, get_unpublished }
+      if (get_published) {
+        params.get_published = get_published === 'true'
+      }
+
+      if (get_unpublished) {
+        params.get_unpublished = get_unpublished === 'true'
+      }
+
+      /* If the user can only read published geos, and all they want is the unpublished geos,
+       * save the trip to the DB and return an array fast.
```

Comment:
sg

Reply:
My argument is that we _should_ throw an error in both cases. Would make diagnosing why nothing is coming back for them much simpler (on both the client end and our end).

Reply:
Had a discussion with Neil, the behavior will become:
- silently filter if no params are supplied and scope is the more restricted one
- throw if params are supplied and scope is restricted

Reply:
We have very similar semantics in `packages/mds-metrics/api.ts`. We still follow the first point, but for the second instead of throwing we simply ignore the requested params in lieu of whatever is attached to the JWT `claim`.

...Perhaps we should change the metrics endpoint to throw as well if the client asks for something they're not allowed to see. @evanxadkins do you have any opinions on this?

Reply:
My general opinion is to filter out anything a user shouldn't see unless they specifically request something prohibited then in that case throw and return`403 - Forbidden`. Most common manifestation is when a user requests a list of things, filter and return what they can see and `200 - Ok` even if the list ends up being empty. If they request something by Id then throw/403 if access isn't allowed.